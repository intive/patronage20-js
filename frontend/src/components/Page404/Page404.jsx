import React from 'react'

import { makeStyles } from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import RefreshIcon from '@material-ui/icons/Refresh'

const useStyles = makeStyles({
  root: {
    height: '100%',
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
  }
})

const handlePageRefresh = () => {
  window.location.reload()
}

export default function Page404 () {
  const classes = useStyles()

  return (
    <Box className={classes.root}>
      <Typography component='h2'>
        <Box style={{ marginBottom: '1rem', fontSize: '3rem', color: '#333' }} textAlign='center'>Coś poszło nie tak...</Box>
      </Typography>
      <Button
        variant='contained'
        color='primary'
        endIcon={<RefreshIcon />}
        onClick={() => { handlePageRefresh() }}
      >
        Odśwież
      </Button>
    </Box>
  )
}
