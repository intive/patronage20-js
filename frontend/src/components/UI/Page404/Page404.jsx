import React from 'react'

import { makeStyles } from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import RefreshIcon from '@material-ui/icons/Refresh'

import { useTranslation } from 'react-i18next'

const useStyles = makeStyles({
  root: {
    height: '100%',
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'column'
  },
  errorMessage: {
    marginBottom: '1rem',
    fontSize: '3rem',
    color: '#333',
    textAlign: 'center'
  }
})

const handlePageRefresh = () => {
  window.location.reload()
}

export default function Page404 () {
  const classes = useStyles()

  const { t } = useTranslation()

  return (
    <Box className={classes.root} data-testid='page-404'>
      <Typography component='h2'>
        <Box className={classes.errorMessage}>{t('something-went-wrong')}.</Box>
      </Typography>
      <Button
        variant='contained'
        color='primary'
        endIcon={<RefreshIcon />}
        onClick={handlePageRefresh}
      >
        {t('refresh')}
      </Button>
    </Box>
  )
}
