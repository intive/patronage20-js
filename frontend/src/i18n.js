import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'

import translations from './common/locales'

const ns = {
  namespaces: ['common', 'dashboard', 'hvac', 'authors']
}

i18n
  // pass the i18n instance to react-i18next.
  .use(initReactI18next)
  // init i18next
  // for all options read: https://www.i18next.com/overview/configuration-options
  .init({
    lng: navigator.language.slice(0, 2),
    fallbackLng: 'en',
    // debug: true,
    ns,
    defaultNS: 'common',
    resources: translations,
    react: {
      useSuspense: false
    },
    interpolation: {
      escapeValue: false // not needed for react as it escapes by default
    }
  })

export default i18n
