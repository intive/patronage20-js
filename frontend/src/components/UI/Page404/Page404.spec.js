/* globals describe, test, expect */

import React from 'react'
import Page404 from './index'
import Button from '@material-ui/core/Button'
import { render, fireEvent } from '@testing-library/react'
import TestRenderer from 'react-test-renderer'
import { I18nextProvider } from 'react-i18next'
import i18n from '../../../i18n'

describe('<Page404 />', () => {
  test('renders Page404 component', () => {
    const root = TestRenderer.create(
      <I18nextProvider i18n={i18n}>
        <Page404 />
      </I18nextProvider>
    )
    expect(root.toJSON()).toMatchSnapshot()
  })

  test('should call function on refresh button click', () => {
    const mockFn = jest.fn()

    const { getByText } = render(
      <Button
        onClick={mockFn}
      >
        Odśwież
      </Button>
    )

    const button = getByText('Odśwież')
    fireEvent.click(button)
    expect(mockFn).toHaveBeenCalled()
  })
})
