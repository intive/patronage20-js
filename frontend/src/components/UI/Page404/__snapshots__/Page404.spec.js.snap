// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page404 /> renders Page404 component 1`] = `
<div
  className="MuiBox-root MuiBox-root-4 makeStyles-root-1"
  data-testid="page-404"
>
  <h2
    className="MuiTypography-root MuiTypography-body1"
  >
    <div
      className="MuiBox-root MuiBox-root-35 makeStyles-errorMessage-2"
    >
      Something went wrong...
      .
    </div>
  </h2>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      REFRESH
      <span
        className="MuiButton-endIcon MuiButton-iconSizeMedium"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root"
          focusable="false"
          role="presentation"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
          />
        </svg>
      </span>
    </span>
  </button>
</div>
`;
